### ISSUE TITLE
Csdid Front End via event_study()

### ISSUE THREAD

**Issue by s3alfisc** [py-econometrics/pyfixest#690]
PyFixest supports a unified front end for three did estimators - twfe, did2s, and linear projections. It would be cool to allow for other edtimstors, potentially implemented in outside packages. 

For example, csdid by @alexanderquispe  implements the C&S estimator. 

We could add csdid as an optional dependency and all users to call it via the event_study frontend.

**Comment by Jayhyung** [comment]
Hi @s3alfisc! I personally want to tackle this issue as I've been through [Callaway and Sant’Anna 2021](https://www.sciencedirect.com/science/article/abs/pii/S0304407620303948) and their package. Did anyone pick this up already(I'm okay to pick other issues If @rafimikail already took this)? Just for the record, are we adding this [csdid](https://github.com/d2cml-ai/csdid) package as a dependency for pyfixest? Thanks always!

**Comment by s3alfisc** [comment]
Hi @Jayhyung, @rafimikail hasn't started on this, so I would happily assign it to you =) The idea was to add `csdid` as an optional runtime dependency so that users do not have to install the package if they don't need it. As a mid-term goal, I was thinking of moving the did module out of pyfixest into a standalone package that functions as a wrapper to multiple did implementations. The counterargument against doing all of this is that we are risking the package do get quite bloated. Not sure what is best. Maybe you have any thoughts on this @Jayhyung?

**Comment by Jayhyung** [comment]
My guess is that it would be certainly better to spin off `did` part as a different package, as users who are interested in `did` module will be the different group of the users. Also CS estimator requires covariate adjustment step using IPW or DR for estimating each group-time ATT, so we also need separate methods(maybe a package or module for this?) for doing this step(`csdid` package imports `drdid` to do this). I guess embedding IPW or DR to the current `pyfixest` codebase will be a bit weird ;). This will be my first time making a standalone package, so let me check on this and study further. So please assign this issue to me and I will bring a bit more concrete plan before I start coding. Thanks!

**Comment by bernardodionisi** [comment]
Or just use [differences](https://github.com/bernardodionisi/differences), which is a fast, scalable, and feature-rich implementation of the various estimation methods discussed in Callaway and Sant'Anna and other papers with that flavor.

**Comment by Jayhyung** [comment]
Hi @bernardodionisi ! This is a great package! I will look into this. Thanks!

**Comment by s3alfisc** [comment]
> By the way, I am also happy to add other methods to differences if you guys are interested in working toward that goal!

Yep, building around differences might also be a good option! We should definitely discuss this =)
